modules/validations/mutually_exclusive_tags.js

```diff
@@ -0,0 +1,97 @@
+import { actionChangeTags } from '../actions/change_tags';
+import { t } from '../core/localizer';
+import { utilDisplayLabel } from '../util';
+import { validationIssue, validationIssueFix } from '../core/validation';
+
+export function validationMutuallyExclusiveTags(/* context */) {
+    const type = 'mutually_exclusive_tags';
+
+    // https://wiki.openstreetmap.org/wiki/Special:WhatLinksHere/Property:P44
+    const tagKeyPairs = [
+        ['noname', 'name'],
+        ['noref', 'ref'],
+        ['nohousenumber', 'addr:housenumber'],
+        ['noaddress', 'addr:housenumber'],
+        ['noaddress', 'addr:housename'],
+        ['noaddress', 'addr:unit'],
+        ['addr:nostreet', 'addr:street']
+    ];
+
+    const validation = function checkMutuallyExclusiveTags(entity /*, graph */) {
+
+        let pairsFounds = tagKeyPairs.filter((pair) => {
+            return (pair[0] in entity.tags && pair[1] in entity.tags);
+        }).filter((pair) => {
+            // noname=no is double-negation, thus positive and not conflicting. We'll ignore those
+            return !((pair[0].match(/^(addr:)?no[a-z]/) && entity.tags[pair[0]] === 'no') ||
+                     (pair[1].match(/^(addr:)?no[a-z]/) && entity.tags[pair[1]] === 'no'));
+        });
+
+        // Addtional:
+        // Check if name and not:name (and similar) are set and both have the same value
```

Comment:
Thanks for humoring my suggestion. This implementation looks pretty good, but the message for this particular situation should be about the _value_ of `not:name` matching the _value_ of `name`, not that both keys exist. After all, the combination of `name=Mi Casa` and `not:name=Su Casa` is valid. You’re currently checking that, but the warning message is a bit misleading.

Moreover, there are two redundant warnings every time `name` matches `not:name`. Either we should keep the preexisting `not:name` validation rule and special-case this code to skip over `not:name`, or we should remove the preexisting `not:name` rule and clarify the messages similar to the original.

<img src="https://github.com/openstreetmap/iD/assets/1231218/c3714a69-6296-4495-b460-75871da8ced4" width="399" alt="ABC has both not:name and name, Building has the mistaken name “ABC”">

Reply:
I moved logic and tests from the `suspicious_name` validator now. You're right I overlooked that `not:name` can be a list of values, semicolon separated. That's also covered now.